### PR TITLE
Fixup ws2812 spi

### PR DESCRIPTION
**Pull request by tzarc**
## Description

Require setting the SPI SCK pin for the WS2812 SPI driver. Fixes up all existing boards to match.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* #12655 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/bt66tech/bt66tech60/config.h [hunk 1]
```diff
@@ -54,7 +54,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define RGB_DI_PIN B15
 #define RGBLED_NUM 9
 #define WS2812_SPI SPID2
-#define WS2812_SPI_MOSI_PAL_MODE 5
+#define WS2812_SPI_SCK_PIN B13
```
Comment (sigprof) [inline keyboards/bt66tech/bt66tech60/config.h:57]: This is definitely not correct — `B13` is in `MATRIX_COL_PINS`. Apparently the F103 chip does not require assigning the SCK function to a physical pin (`platforms/chibios/STM32_F103_STM32DUINO/board/board.h` does not seem to enable any SCK pins). Also `SPI2_SCK` is not even mappable to any other pin than `B13` on F103.

File: keyboards/cannonkeys/practice65/config.h [hunk 2]
```diff
@@ -54,6 +54,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define RGB_DI_PIN B15
 #define RGBLED_NUM 20
 #define WS2812_SPI SPID2
+#define WS2812_SPI_SCK_PIN B13
```
Comment (sigprof) [inline keyboards/cannonkeys/practice65/config.h:57]: Again, `B13` is in `MATRIX_COL_PINS` on this board, so this change is definitely not correct for this hardware.

**Review by sigprof — COMMENTED** [review]
Did not try this on any real hardware, but it looks like changing the config for F103 and F303 this way would break some existing boards (and in some cases there is even no free pin that could be used for SCK). So the driver configuration for SCK probably needs to be made optional with no default, and `config.h` files for F072 boards need to be updated (or, alternatively, the default might be made chip-specific and enabled only for F072).

File: keyboards/cannonkeys/practice65/config.h (hunk 2 above)
Comment (awkannan) [reply keyboards/cannonkeys/practice65/config.h:57]: you can leave this one as is - we don't sell Practice65s anymore (only V2).

this is f103 powered, and worked without SCK set I think.

File: drivers/chibios/ws2812_spi.c [hunk 3]
```diff
@@ -16,22 +16,36 @@
 #    define WS2812_SPI_MOSI_PAL_MODE 5
 #endif
 
+#ifndef WS2812_SPI_SCK_PIN
+#    define WS2812_SPI_SCK_PIN B3
+#endif
+
+#ifndef WS2812_SPI_SCK_PAL_MODE
+#    define WS2812_SPI_SCK_PAL_MODE 5
+#endif
```
[the remark this answers is not in the document]
Comment (tzarc) [reply drivers/chibios/ws2812_spi.c]: Limited the scope to F072-based boards, for now. Also made the pin optional -- as you recommended the `palSetLineMode()` is now only invoked if it's specified.

**Comment by bcat** [comment]
I wasn't sure if you still wanted someone with relevant hardware to verify this, but I can confirm this fixes RGB on my CannonKeys Instant60 (which stopped working after I pulled in many months of QMK updates to my fork and recompiled).

**Comment by tzarc** [comment]
Had a few more confirmations that this fixes boards on the CannonKeys discord this morning.

**Comment by tzarc** [comment]
If we find others still broken for other MCUs, will followup with another PR.

**Comment by yiancar** [comment]
Sorry just saw this, Could we please enchanse the documentation a bit with the explanation from https://github.com/qmk/qmk_firmware/issues/12655#issuecomment-844104659 ? Especially the bit that now we know for a fact that the SCK pin should remain unused. 

It is a pretty interesting find @sigprof, do you have any links describing the behavior of the SPI controller ?

**Comment by sigprof** [comment]
These detail do not seem to be described in the official documentation, and the only note mentioning the SPI dependency on having the SCK pin assigned is https://community.st.com/s/question/0D53W00000XiwolSAB/stm32h7a3-spi-not-working-if-sck-disabled (but that's from an discussion of a completely different chip with a much newer SPI core):

> In the "normal" STM32 SPI the receiver is clocked by a feedback = input from the SCK pin, so while Tx works, Rx won't without enabling the SCK pin.

There is also this document, which list the SPI peripheral versions for various STM32 chips: https://www.st.com/resource/en/application_note/dm00725181-enhanced-methods-to-handle-spi-communication-on-stm32-devices-stmicroelectronics.pdf

That document claims that STM32F1, STM32F2, STM32F4, STM32L0,STM32L1 have SPI 1.2.x, and STM32F0, STM32F3, STM32F7, STM32L4, STM32L5, STM32WB, STM32WL have SPI 1.3.x; but it also does not completely match the behavior difference with SCK — F072 definitely requires SCK, some L4 chips apparently have the same requirement, but F303, which is placed in the same category, apparently is able to work without SCK.

**Comment by yiancar** [comment]
Cool thank you for the info!

**Pull request merged**